src/features/ci-cd/hooks/track-latest-release.ts

```diff
@@ -1,20 +1,29 @@
 import * as _ from 'lodash';
 
-import { sbvrUtils, hooks } from '@balena/pinejs';
-
-const updateLatestRelease = async (
-	id: number,
-	{ request, api }: hooks.HookArgs,
-) => {
-	// We only track builds that are successful
-	if (request.values.status !== 'success') {
+import { hooks } from '@balena/pinejs';
+const releaseStateFields = [
+	'status',
+	'is_passing_tests',
+	'release_type',
+	'is_invalidated',
+];
+const updateLatestRelease = async ({ request, api }: hooks.HookArgs) => {
+	if (!releaseStateFields.some((field) => field in request.values)) {
```

Comment:
These fields have defaults on creation right, which means they might be specified but still potentially end up with the correct values to become the new default - is that handled correctly?

Reply:
@page- i may just be fried but i don't quite follow what you mean here, can you rephrase please?

Reply:
So this is called in the `POST` hook, at which point the request may omit fields and therefore `field in request. values === false` however those fields have defaults in the database which means that they have an implicit value, eg `is_passing_tests` will have the implicit value `true`. I think it'd be fine in practice as I think it'd be required to specify at least `status` on create in order to alter the latest release however it'd be better to be safe to it by design (or at least by comment)

Reply:
this function is called both in the `POST` and `PATCH` hooks, which is how releases are created aiui (first a stub is POSTed, then PATCHed when more information is available). The default values provided by the DB are set in such a way that with no extra metadata they will be set as the latest release (failing open rather than closed, in a way). since this is already the expected behavior when building, i believe this is a cleaner change than making these metadata opt-out. am i still missing something here? if all the fields were to be omitted, there couldn't be a `status==success` since that's not provided as a default, and therefore the latest release couldn't be updated.

Reply:
Yes that's what I'm saying, it happens to be the case that this is safe because the defaults do not match a valid latest results, but if those defaults were changed such that they did create a valid latest release we would short-circuit when we should not. Hence ideally I want the design to enforce that we're safe even if those defaults change, or at a minimum document that the defaults must not create a a valid latest release

Reply:
sorted out in https://www.flowdock.com/app/rulemotion/resin-tech/threads/Vgitvd6OIOi5BoAJUq5ClqqkvLk